support/k8s/k8s-workload-registrar/mode-crd/controllers/pod_controller_test.go

```diff
@@ -142,6 +148,212 @@ func (s *PodControllerTestSuite) TestPodLabel() {
 	}
 }
 
+// TestIdentityTemplate checks the various formats of the SPIFFE ID provided via IdentityTemplate.
+func (s *PodControllerTestSuite) TestIdentityTemplate() {
+	tests := []struct {
+		identityTemplate      string
+		identityTemplateLabel string
+		labelValue            string
+		context               map[string]string
+		expectedSVID          string
+		uid                   string
+		err                   string
+		spiffeIDCount         int
+	}{
+		// this section is testing error conditions:
+		{
+			// invalid template format:
+			identityTemplate: "invalid/",
+			err:              "invalid SVID, ends with /",
+		},
+		{
+			// missing Context map:
+			identityTemplate: "region/{{.Context.Region}}",
+			err:              "template references a value not included in context map",
+		},
+		{
+			// invalid Context reference:
+			context: map[string]string{
+				"Region":      "EU-DE",
+				"ClusterName": "MYCLUSTER",
+			},
+			identityTemplate: "error/{{.Context.XXXX}}",
+			err:              "template references a value not included in context map",
+		},
+		{
+			// invalid Pod reference:
+			identityTemplate: "region/{{.Pod.XXXX}}",
+			err:              "can't evaluate field XXXX ",
+		},
+
+		// this section is testing the identity_template_label:
+		{
+			// label requested, pod not labeled
+			identityTemplate:      SimpleTemplate,
+			identityTemplateLabel: IdentityLabel,
+			spiffeIDCount:         0,
+		},
+		{
+			// label requested, pod labeled 'false'
+			identityTemplate:      SimpleTemplate,
+			identityTemplateLabel: IdentityLabel,
+			labelValue:            "false",
+			spiffeIDCount:         0,
+		},
+		{
+			// label not set
+			identityTemplate: DefaultTemplate,
+			expectedSVID:     fmt.Sprintf("ns/%s/sa/%s", PodNamespace, PodServiceAccount),
+			spiffeIDCount:    1,
+		},
+		{
+			// label empty
+			identityTemplate:      DefaultTemplate,
+			identityTemplateLabel: "",
+			expectedSVID:          fmt.Sprintf("ns/%s/sa/%s", PodNamespace, PodServiceAccount),
+			spiffeIDCount:         1,
+		}, 
+		{
+			// label requested, pod labeled
+			identityTemplate:      DefaultTemplate,
+			identityTemplateLabel: IdentityLabel,
+			labelValue:            "true",
+			expectedSVID:          fmt.Sprintf("ns/%s/sa/%s", PodNamespace, PodServiceAccount),
+			spiffeIDCount:         1,
+		},
+		{
+			// label empty, pod labeled false
+			identityTemplate:      DefaultTemplate,
+			identityTemplateLabel: "",
+			labelValue:            "false",
+			expectedSVID:          fmt.Sprintf("ns/%s/sa/%s", PodNamespace, PodServiceAccount),
+			spiffeIDCount:         1,
+		},
+
+		// this section is testing identity template formatting:
+		{
+			identityTemplate: DefaultTemplate,
+			expectedSVID:     fmt.Sprintf("ns/%s/sa/%s", PodNamespace, PodServiceAccount),
+			spiffeIDCount:    1,
+		},
+		{
+			identityTemplate: "ns/{{.Pod.Namespace}}/sa/{{.Pod.ServiceAccount}}",
+			expectedSVID:     "ns/" + PodNamespace + "/sa/" + PodServiceAccount,
```

Comment:
nit: since all the other tests are building the string with `fmt.Sprintf()`, can we do the same here just to be consistent?